packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts

```diff
@@ -579,4 +639,53 @@ export class GutenbergEditorPage {
 		}
 		await this.editorToolbarComponent.openDesktopPreview( 'Desktop' );
 	}
+
+	/* Misc */
+
+	/**
+	 * Leave the editor to return to the Calypso dashboard.
+	 *
+	 * On desktop sized viewport, this method first opens the editor navigation
+	 * sidebar, then clicks on the `Dashboard` link.
+	 *
+	 * On mobile sized viewport, this method clicks on Navbar > My Sites.
+	 *
+	 * The resulting page can change based on where you come from, and the viewport. Either way, the resulting landing spot
+	 * will have access to the Calyspo sidebar, allowing navigation around Calypso.
+	 */
+	async exitEditor(): Promise< void > {
+		await this.editorNavSidebarComponent.openSidebar();
+
+		// There are three different places to return to,
+		// depending on how the editor was entered.
+		const navigationPromise = Promise.race( [
+			this.page.waitForNavigation( { url: '**/home/**' } ),
+			this.page.waitForNavigation( { url: '**/posts/**' } ),
+			this.page.waitForNavigation( { url: '**/pages/**' } ),
+		] );
+		const actions: Promise< unknown >[] = [ navigationPromise ];
+
+		if ( envVariables.VIEWPORT_NAME === 'mobile' ) {
+			// Mobile viewports do not use an EditorNavSidebar.
+			// Instead, the regular NavBar is used, and the
+			// `My Sites` button exits the editor.
+			const navbarComponent = new NavbarComponent( this.page );
+			actions.push( navbarComponent.clickMySites() );
+		} else {
+			actions.push( this.editorNavSidebarComponent.exitEditor() );
+		}
+
+		// Perform the actions and resolve promises.
+		await Promise.all( actions );
+	}
+
+	/**
+	 * Checks whether the editor has any block warnings/errors displaying.
+	 *
+	 * @returns True if there are block warnings/errors, false otherwise.
+	 */
+	async editorHasBlockWarnings(): Promise< boolean > {
```

Comment:
As above, this is an reorganization of the method order and nothing was changed.